dcm4che-net-audit/src/main/java/org/dcm4che3/net/audit/AuditLogger.java

```diff
@@ -743,6 +756,61 @@ public void setRetryInterval(int interval) {
         this.retryInterval = interval;
     }
 
+    /**
+     * Get size of ring buffer (in bytes) used to write audit messages
+     * @return ring buffer size in bytes
+     */
+    public int getRingBufferSize() {
+        return ringBufferSize;
+    }
+
+    /**
+     * Set size of ring buffer used to write audit messages, based on system property
+     * {@value RING_BUFFER_SIZE}
+     * Enforce minimum size {@value MINIMUM_RING_BUFFER_SIZE} and size as power of 2
+     */
+    private static int setRingBufferSize() {
+        return setRingBufferSize(Integer.parseInt(System.getProperty(RING_BUFFER_SIZE,
+                String.valueOf(DEFAULT_RING_BUFFER_SIZE))));
+    }
+
+    /**
+     * Set size of ring buffer used to write audit messages.
+     * Enforce minimum size {@value MINIMUM_RING_BUFFER_SIZE} and size as power of 2
+     * @param bufferSize proposed ring buffer size
+     * @return ring buffer size as power of 2
+     */
+    public static int setRingBufferSize(int bufferSize) {
+        int adjustedBufferSize = (bufferSize < MINIMUM_RING_BUFFER_SIZE ?
+                MINIMUM_RING_BUFFER_SIZE : getPowerOfTwo(bufferSize));
+
+        if (adjustedBufferSize != bufferSize) {
+            LOG.warn("Ring buffer size {} adjusted from {} to {} as nearest (higher) power of 2 value", bufferSize,
+                    adjustedBufferSize);
+        }
+
+        ringBufferSize = adjustedBufferSize;
+        return adjustedBufferSize;
+    }
+
+    /** Get the number that's the smallest power of two greater than or equal to number
+     *
+     * @param value value to check as power of two
+     * @return number that's the smallest power of two greater than or equal to number
+     */
+    private static int getPowerOfTwo(int value) {
+        if (value <= 1) {
+            return 2;
+        };
+
+        int next = Integer.highestOneBit(value);
+        if (value == next) {
```

Comment:
You don't really need this check. Just remove the if clause entirely and use the following return statement.